### PR TITLE
Add a fake git server image for e2e test

### PR DESCRIPTION
**Pull request by resouer**
Will be used in #18445, cc @thockin

### CODE REVIEW

**Comment by ixdy** [comment]
I'll let @pmorie or @thockin review this one since it goes along with #18445.

File: test/images/fakegitserver/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,18 @@
+# Copyright 2016 Google Inc. All rights reserved.
```
Comment (ixdy) [inline test/images/fakegitserver/Dockerfile]: this should probably be Kubernetes Authors

**Comment by resouer** [comment]
@k8s-bot unit test this

File: test/images/fakegitserver/gitserver.go [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+func hello(w http.ResponseWriter, r *http.Request) {
+	io.WriteString(w, "I am a fake git server")
+
+}
+
+func main() {
+	http.HandleFunc("/", hello)
+	http.ListenAndServe(":8000", nil)
+
```
Comment (pmorie) [inline test/images/fakegitserver/gitserver.go]: Remove this newline and add a comment explaining that you get an empty repository when you clone from this server.

**Comment by pmorie** [comment]
One nit, then LGTM.

**Comment by resouer** [comment]
Thanks, nit fixed

**Comment by pmorie** [comment]
Thanks, @resouer

**Comment by pmorie** [comment]
@thockin What's the process for getting this pushed to gcr.io?

**Comment by thockin** [comment]
Pushed gcr.io/google_containers/fakegitserver:0.1

**Pull request merged**